bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/inbox/InboxFilterCriteria.java

```diff
@@ -26,61 +27,137 @@
     private final String bindingId;
     private final ThingTypeUID thingTypeUID;
     private final ThingUID thingUID;
+    private final String representationValue;
```

Comment:
The problem with InboxFilterCritearia is, that they are rather old. As I didn't want to break backward compatibility, I did not remove the constructors, but made them deprecated. Not sure, if changing thic class into something more Java 8 like is not breaking the compatibility. But I agree, it's a bit ugly. Will think about it.